src/main/java/net/atos/entng/rbs/service/BookingServiceSqlImpl.java

```diff
@@ -1021,4 +1030,119 @@ public void getBooking(final String bookingId, final Handler<Either<String, Json
 	}
 
+	public Future<JsonObject> getBooking(final String bookingId) {
+		Promise<JsonObject> promise = Promise.promise();
+		retrieve(bookingId, FutureHelper.handlerJsonObject(promise));
+		return promise.future();
+	}
+
+	public Future<JsonObject> deleteBooking(String id, UserInfos user) {
+		Promise<JsonObject> promise = Promise.promise();
+		delete(id, user, FutureHelper.handlerJsonObject(promise));
+		return promise.future();
+	}
+
+	/**
+	 * Iterates through an array of bookings to save them
+	 * The index of the resource ids must be the same as the corresponding booking
+	 * @param bookingIds {@link List<Integer>} the array of resource ids
+	 * @param user {@link UserInfos} the user
+	 * @return {@link Future<JsonObject>} an array of saved bookings
+	 */
+	@Override
+	public Future<List<JsonObject>> checkRightsAndDeleteBookings(final List<Integer> bookingIds, final UserInfos user) {
+		Promise<List<JsonObject>> promise = Promise.promise();
+
+		List<Future<JsonObject>> bookingsFuture = new ArrayList<>();
+
+		for (Integer bookingId : bookingIds) {
+			String bookingStringId = String.valueOf(bookingId);
+			bookingsFuture.add(checkRightsAndDeleteBooking(bookingStringId, user));
+		}
+
+		FutureHelper.all(bookingsFuture)
+			.onSuccess((res) -> {
+				promise.complete(bookingsFuture.stream().map(Future::result).collect(Collectors.toList()));
+			})
+			.onFailure((err) -> {
+				String message = String.format("[Rbs@%s::createBookings]: an error has occurred while saving bookings: %s",
+						this.getClass().getSimpleName(), err.getMessage());
+				log.error(message);
+				promise.fail(err.getMessage());
+			});
+
+
+		return promise.future();
+	}
+
+
+	/**
+	 * This methods checks if the user has the right to delete the booking and if so deletes it
+	 * All cases lead to completed promises in order for checkRightsAndDeleteBookings() to gather all the booking deletion info
+	 * and send it back to Calendar Module using event bus response.
+	 * Example of completed promise :
+	 * {
+	 *     "id": "000",
+	 *     "status": "NOT DELETED",
+	 *     "message": "Unauthorized"
+	 * }
+	 * @param bookingStringId {@link String} the id of the booking
+	 * @param user {@link UserInfos} the user
+	 * @return {@link Future<JsonObject>} (see example above)
+	 */
+	Future<JsonObject> checkRightsAndDeleteBooking(String bookingStringId, UserInfos user) {
+		Promise<JsonObject> promise = Promise.promise();
+		this.getBooking(bookingStringId)
+				.compose(booking -> {
+					String resourceId = booking.getLong(Field.RESOURCE_ID).toString();
+					String method = BookingController.class.getName() + "|deleteBooking";
+					Binding binding = new Binding(HttpMethod.DELETE, Pattern.compile(""), method, ActionType.RESOURCE);
+					return new TypeAndResourceAppendPolicy().authorize(resourceId, bookingStringId, binding, user);
+				})
+				.onSuccess(hasRight -> {
+					if (Boolean.TRUE.equals(hasRight)) {
+						promise.complete(new JsonObject()
```

Comment:
temporary, will be deleted in next ticket  https://entsupport.gdapublic.fr/browse/RBS-135